core-graphics/src/image.rs

```diff
@@ -121,6 +122,10 @@ impl CGImageRef {
         };
         data_provider.copy_data()
     }
+
+    pub fn cropped(&self, rect: CGRect) -> CGImage {
+        unsafe { CGImage::from_ptr(CGImageCreateWithImageInRect(self.as_ptr(), rect)) }
```

Comment:
This can return null if the rect defines an area that is not in the image: https://developer.apple.com/documentation/coregraphics/1454683-cgimagecreatewithimageinrect .